LEAF_Request_Portal/sources/Email.php

```diff
@@ -849,10 +856,15 @@ function attachApproversAndEmail(int $recordID, int $emailTemplateID, mixed $log
                         $resStep = $this->portal_db->prepared_query($strSQL, $varsStep);
 
                         $resGroupID = $form->getIndicator($resStep[0]['indicatorID_for_assigned_groupID'], 1, $recordID);
-                        $groupID = $resGroupID[$resStep[0]['indicatorID_for_assigned_groupID']]['value'];
-
-                        if ($groupID > 0) {
-                            $this->addGroupRecipient($groupID);
+                        
+                        // groupid is required to move forward, make sure this exists before continuing. 
+                        // This can be a result of user not setting a group in form field
+                        if(is_array($resGroupID) && !empty($resGroupID[$resStep[0]['indicatorID_for_assigned_groupID']])){
```

Comment:
I'm thinking this needs an else the returns an error to let the user know why it failed.

Reply:
Dropped in a trigger error, is there anything we really need to display there? Having a hard time imagining what I should do have here.